org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -1394,12 +1400,18 @@ protected boolean load (String uuID,String trxName)
 		{
 			if (i != 0)
 				sql.append(",");
-			String columnSQL = p_info.getColumnSQL(i);
-			if (!p_info.isVirtualColumn(i))
+			String columnSQL; 
+			if (p_info.isVirtualColumn(i))
+			{
+				// initialize with NULL - values will be lazy-loaded by getters
+				columnSQL = "NULL AS " + p_info.getColumnName(i);
```

Comment:
I think this needs to be changed to DB.getDatabase().quoteColumnName(p_info.getColumnName(i)) since p_info.getColumnName(i) could be a db reserved word.

Reply:
Yes, or it could simply contain spaces (not sure how the model generator would react to that, though). I deliberately tried to leave the original code untouched, which is:

```
			String columnSQL = p_info.getColumnSQL(i);
			if (!p_info.isVirtualColumn(i))
			{
				columnSQL = DB.getDatabase().quoteColumnName(columnSQL);
			}
			sql.append(columnSQL);	//	Normal and Virtual Column
```

I doesn't seem to make sense to treat virtual columns differently, especially when their names come from metadata.

Maybe a better approach could be to make this quoting at `POInfo.getColumnSQL(int)`

We'll make that change during the day. Thanks for reviewing.

Reply:
This could be probably out of the scope of this issue, but quoting the name assigned to virtual columns should be also performed at `POInfo.buildSelect(boolean, boolean)`, which is currently used by `Query` in most cases. This could be an indicator that this is not a current issue with iDempiere core. In any case, if developers use a column name that is a reserved keyword they should notice and fix that problem on the spot.